imt/doc.go

```diff
@@ -0,0 +1,2 @@
+// Package imt provides an Incremental Merkle Tree
+package imt
```

Comment:
A `doc.go` is good practice if it's not obvious what the package does. Just reading `imt` was not explicit enough for me. I had to dig around and figure out what this package provided.